src/CodeGenHelpers/ConstructorBuilder.cs

```diff
@@ -144,25 +134,15 @@ public ConstructorBuilder WithThisCall(IMethodSymbol baseConstructor)
                         _parameters.Add(param.Type.Name, param.Name);
                 }
 
-                var output = _parameters.Where(x => baseConstructor.Parameters.Any(p => p.Type.Name == x.Key || p.Type.GetFullMetadataName() == x.Key)).Select(x => x.Value);
+                var output = _parameters.Where(x => baseConstructor.Parameters.Any(p => p.Type.Name == x.Key || p.Type.GetFullMetadataName() == x.Key)).Select(x => x.Key);
                 return $": this({string.Join(", ", output)})";
             };
             return this;
         }
 
-        public ConstructorBuilder WithBaseCall(Dictionary<string, string> parameters)
+        public ConstructorBuilder WithBaseCall(params string[] parameters)
```

Comment:
I'd rather that you introduce an override than change the signature outright. You're also outright breaking expected behavior here where the base parameters should be checking for and being added to the parameters in the main constructor... so you might have:

```cs
// expected
public class Foo : Bar
{
    public Foo(string someValue, Baz baz) : base(baz)
    {
    }
}

// builder
constructorBuilder
    .AddParameter("string", "someValue")
    .WithBaseCall(new Dictionary<string, string> { { "Baz", "baz" } });
```